packages/cpp/ArmoniK.Api.Worker/source/Worker/TaskHandler.cpp

```diff
@@ -402,14 +402,16 @@ std::map<std::string, std::string> API_WORKER_NAMESPACE::TaskHandler::getDataDep
  *
  * @return armonik::api::grpc::v1::TaskOptions
  */
-armonik::api::grpc::v1::TaskOptions API_WORKER_NAMESPACE::TaskHandler::getTaskOptions() { return task_options_; }
+const armonik::api::grpc::v1::TaskOptions &API_WORKER_NAMESPACE::TaskHandler::getTaskOptions() const {
```

Comment:
Why do you return a gRPC object? Don't you have a `TaskOptions` class in the API?

Reply:
TaskOptions in in the SDK, not Api

Reply:
Ok. I think it would be good to have the TaskOptions type decorrelated from gRPC in the API lib. Not for this PR anyway.